staging/src/k8s.io/apiserver/pkg/endpoints/handlers/apply.go

```diff
@@ -19,162 +19,66 @@ package handlers
-	// Make sure we have the gvk set on the object.
-	(&unstructured.Unstructured{Object: patch}).SetGroupVersionKind(p.kind)
-
-	j, err := json.Marshal(patch)
-	if err != nil {
-		return fmt.Errorf("failed to serialize json: %v", err)
-	}
-
-	accessor, err := meta.Accessor(dst)
-	if err != nil {
-		return fmt.Errorf("couldn't get accessor: %v", err)
-	}
-	m := accessor.GetManagedFields()
-	if m == nil {
-		m = make(map[string]metav1.VersionedFieldSet)
-	}
-	// TODO: save the managedFields correctly
-	m[workflow] = metav1.VersionedFieldSet{
-		APIVersion: string(j),
-	}
-	accessor.SetManagedFields(m)
-	return nil
-}
-
-func (p *applyPatcher) applyPatchToCurrentObject(currentObject runtime.Object) (runtime.Object, error) {
-	current, err := p.convertCurrentVersion(currentObject)
-	if err != nil {
-		return nil, fmt.Errorf("failed to convert current object: %v", err)
-	}
-
-	lastIntent, err := p.extractLastIntent(currentObject, workflowId)
-	if err != nil {
-		return nil, fmt.Errorf("failed to extract last intent: %v", err)
-	}
-	newIntent, err := p.getNewIntent()
-	if err != nil {
-		return nil, fmt.Errorf("failed to get new intent: %v", err)
-	}
-
-	element, err := parse.CreateElement(lastIntent, newIntent, current, p.models.LookupModel(""))
-	if err != nil {
-		return nil, fmt.Errorf("failed to parse elements: %v", err)
-	}
-	result, err := element.Merge(strategy.Create(strategy.Options{}))
-	if err != nil {
-		return nil, fmt.Errorf("failed to merge elements: %v", err)
+		return nil, err
 	}
 
-	output, err := p.convertResultToUnversioned(result)
+	newObj, err := p.converter.TypedToObject(outputTyped)
 	if err != nil {
-		return nil, fmt.Errorf("failed to convert merge result: %v", err)
+		return nil, fmt.Errorf("failed to convert output typed to object: %v", err)
 	}
 
-	if err := p.saveNewIntent(newIntent, workflowId, output); err != nil {
-		return nil, fmt.Errorf("failed to save last intent: %v", err)
+	if err := apply.EncodeObjectManagedFields(newObj, managed); err != nil {
+		return nil, err
 	}
 
-	// TODO(apelisse): Check for conflicts with other managedFields
-	// and report actionable errors to users.
-
-	return output, nil
+	return newObj, nil
 }
 
 func (p *applyPatcher) createNewObject() (runtime.Object, error) {
-	original := p.restPatcher.New()
-	objToCreate, gvk, err := p.codec.Decode(p.patchBytes, &p.kind, original)
-	if err != nil {
-		return nil, transformDecodeError(p.typer, err, original, gvk, p.patchBytes)
-	}
-	if gvk.GroupVersion() != p.kind.GroupVersion() {
-		return nil, errors.NewBadRequest(fmt.Sprintf("the API version in the data (%s) does not match the expected API version (%v)", gvk.GroupVersion().String(), p.kind.GroupVersion().String()))
-	}
+	return p.applyPatchToCurrentObject(p.restPatcher.New())
+}
 
-	newIntent, err := p.getNewIntent()
-	if err != nil {
-		return nil, fmt.Errorf("failed to get new intent: %v", err)
-	}
+type failConverter struct{}
```

Comment:
it looks like you're using the real converter below, can you remove this one and use the real one everywhere?